src/writeckpt.cpp

```diff
@@ -352,7 +352,7 @@ static void mtcp_write_non_rwx_and_anonymous_pages(int fd, Area *orig_area)
    * code and that should reset the /proc/self/maps files to its original
    * condition.
    */
-  JASSERT(orig_area->name[0] == '\0');
+  JASSERT(orig_area->name[0] == '\0' || (strcmp(orig_area->name, "[heap]") == 0));
 
```

Comment:
@gc00 Adding stack sections too require the same change in above assert too right ? 
JASSERT(orig_area->name[0] == '\0' || (strcmp(orig_area->name, "[heap]") == 0) ||
                 (strcmp(orig_area->name, "[stack]"== 0) || (orig_area->name, "[stack:XXX]"))